app/code/Bulbulatory/Recomendations/Model/RecommendationRepository.php

```diff
@@ -0,0 +1,113 @@
+<?php
+ 
+namespace Bulbulatory\Recomendations\Model;
+ 
+use Magento\Framework\Api\SearchCriteriaInterface;
+use Magento\Framework\Api\SortOrder;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Bulbulatory\Recomendations\Api\Data\RecommendationInterface;
+use Bulbulatory\Recomendations\Api\Data\RecommendationSearchResultInterface;
+use Bulbulatory\Recomendations\Api\Data\RecommendationSearchResultInterfaceFactory;
+use Bulbulatory\Recomendations\Api\RecommendationRepositoryInterface;
+use Bulbulatory\Recomendations\Model\ResourceModel\Recommendation\CollectionFactory as RecommendationCollectionFactory;
+use Bulbulatory\Recomendations\Model\ResourceModel\Recommendation\Collection;
+ 
+class RecommendationRepository implements RecommendationRepositoryInterface
+{
+    /**
+     * @var RecommendationFactory
+     */
+    private $recommendationFactory;
+ 
+    /**
+     * @var RecommendationCollectionFactory
+     */
+    private $recommendationCollectionFactory;
+ 
+    /**
+     * @var RecommendationSearchResultInterfaceFactory
+     */
+    private $searchResultFactory;
+ 
+    public function __construct(
+        RecommendationFactory $recommendationFactory,
+        RecommendationCollectionFactory $recommendationCollectionFactory,
+        RecommendationSearchResultInterfaceFactory $recommendationSearchResultInterfaceFactory
+    ) {
+        $this->recommendationFactory = $recommendationFactory;
+        $this->recommendationCollectionFactory = $recommendationCollectionFactory;
```

Comment:
`$this->recommendationCollectionFactory` nie jest używane nigdzie poza konstruktorem, więc można je wyrzucić.